index_alias_impl.go

```diff
@@ -573,11 +576,20 @@ type asyncSearchResult struct {
 type preSearchFlags struct {
 	knn      bool
 	synonyms bool
+	bm25     bool // needs presearch for this too
 }
 
-// preSearchRequired checks if preSearch is required and returns a boolean flag
-// It only allocates the preSearchFlags struct if necessary
-func preSearchRequired(req *SearchRequest, m mapping.IndexMapping) (*preSearchFlags, error) {
+func isBM25Enabled(m mapping.IndexMapping) bool {
```

Comment:
Let's refactor this to `scoringModel(..)` which returns the scoring model to use instead.

Reply:
On second thought, perhaps this is fine as is - given that IndexMappingImpl's ScoringModel is exported.